cpu/atmega_common/periph/timer.c

```diff
@@ -73,6 +73,23 @@ static ctx_t ctx[] = {
 #endif
 };
 
+static unsigned _oneshot;
+
+static inline void set_oneshot(tim_t tim, int chan)
```

Comment:
Why are you not using `uintX_t` types here, but `unsigned` and `int`. Any specific reason for that?

Reply:
This is actually consistent with [the rest of the API](http://api.riot-os.org/group__drivers__periph__timer.html).

8 bit platforms seem to have timers of at most 16 bit width, while 32 bit platforms seem to have at most 32 bit width. An `unsigned` for the timer value thus is big enough for both. On 8 bit platforms an `unsigned` however is faster than an `uint32_t`. So maybe we should evaluate carefully if there is actually an advantage for `uintX_t` here, before changing the API.

Reply:
For the `unsigned flags` in `timer_set_periodic()`, an `uint<NUM>_t` would however make sense. An `uint8_t` would also allow all architectures to pass the value as a single register (looking at AVR).

Reply:
It's an `uint8_t` now.